modules/tasks/tasks/src/main/java/org/motechproject/tasks/web/ActivityController.java

```diff
@@ -95,17 +95,19 @@ public TaskActivityRecords getAllActivities(GridSettings settings) {
     @ResponseBody
     public TaskActivityRecords getTaskActivities(@PathVariable Long taskId, GridSettings settings) {
         if (settings != null) {
-            QueryParams params = new QueryParams(settings.getPage(), settings.getRows(), new Order("date", Order.Direction.DESC));
+            QueryParams params = settings.isLastExecution() ? new QueryParams(settings.getPage(), 1, new Order("date", Order.Direction.DESC)) :
+                    new QueryParams(settings.getPage(), settings.getRows(), new Order("date", Order.Direction.DESC));
+
             Set<TaskActivityType> types = settings.getTypesFromString();
             Range<DateTime> dateTimeRange = settings.convertToDateRange(settings.getDateTimeFrom(), settings.getDateTimeTo());
             List<TaskActivityDto> activities;
             long count;
             if (dateTimeRange != null) {
                 activities = taskWebService.getTaskActivities(taskId, types, dateTimeRange, params);
-                count = activityService.getTaskActivitiesCount(taskId, types, dateTimeRange);
+                count = settings.isLastExecution() ? 1 : activityService.getTaskActivitiesCount(taskId, types, dateTimeRange);
```

Comment:
If condition about the last execution returns true, then it is 1, because this piece of code is for one, specified task, in another case will be invoke counter from activityService.

Edit:
I found better solution for this problem. I will push this with corrections.